source/include/core_mqtt_agent_config_defaults.h

```diff
@@ -0,0 +1,119 @@
+/*
+ * coreMQTT Agent v1.1.0
+ * Copyright (C) 2021 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file core_mqtt_agent_config_defaults.h
+ * @brief This represents the default values for the configuration macros
+ * for the MQTT-Agent library.
+ *
+ * @note This file SHOULD NOT be modified. If custom values are needed for
+ * any configuration macro, a core_mqtt_agent_config.h file should be provided to
+ * the MQTT-Agent library to override the default values defined in this file.
+ * To use the custom config file, the MQTT_AGENT_USE_CUSTOM_CONFIG
+ * preprocessor macro SHOULD be set.
+ */
+
+#ifndef CORE_MQTT_AGENT_CONFIG_DEFAULTS_H_
+#define CORE_MQTT_AGENT_CONFIG_DEFAULTS_H_
+
+/* *INDENT-OFF* */
+#ifdef __cplusplus
+    extern "C" {
+#endif
+/* *INDENT-ON* */
+
+/* MQTT_AGENT_USE_CUSTOM_CONFIG allows building the MQTT library
+ * without a custom config. If a custom config is provided, the
+ * MQTT_AGENT_USE_CUSTOM_CONFIG macro should be defined. */
+#ifdef MQTT_AGENT_USE_CUSTOM_CONFIG
```

Comment:
For consistency sake - should this not be MQTT_AGENT_DO_NOT_USE_CUSTOM_CONFIG https://github.com/FreeRTOS/coreMQTT/blob/10d85cb9f96f680072c474bef415fc76bf673642/source/include/core_mqtt_config_defaults.h#L47